tests/e2e_decorrelation/test_error_cases.py

```
@@ -0,0 +1,715 @@
+        bound_plan = binder.bind(logical_plan)
+
+        # Verify plan structure
+        assert_plan_structure(decorrelated_plan, {})
+        results = execute_and_fetch_all(executor, decorrelated_plan)
+        # with pytest.raises(DecorrelationError, match="multiple rows"):
+        #     decorrelated_plan = decorrelator.decorrelate(bound_plan)
+
+        # For now, just verify it's decorated (may fail at execution)
+        decorrelated_plan = decorrelator.decorrelate(bound_plan)
+
+    def test_scalar_subquery_multiple_columns_error(self, catalog, setup_test_data):
```

Ruff (0.14.5)

Unused method argument: `setup_test_data`

(ARG002)